Assignment 1/src/main/java/com/nuclei/assignment/service/itemadder/ItemAdderImpl.java

```diff
@@ -0,0 +1,205 @@
+    while (addMoreToItems.equals(StringConstantsUtils.CONFIRMATION)) {
+      try {
+        String[] input = rawData;
+        if (!rawDataAvailable) {
+          System.out.println(StringConstantsUtils.INSERT_ITEM_DETAILS);
+          input = scanner.nextLine().split(" ");
+        }
+        final ItemEntity item = createItem(input);
+        logger.info(String.format("Created %s", item));
+        items.add(item);
+      } catch (Exception exception) {
+        System.out.println(exception.getMessage());
+      } finally {
+        rawDataAvailable = false;
+        System.out.println(StringConstantsUtils.DIVIDER);
+        System.out.println(StringConstantsUtils.ADD_MORE_ITEMS_QUERY);
+  
+        addMoreToItems = scanner.nextLine().toLowerCase(Locale.ROOT);
+      }
+    }
+    return items;
+  }
+  
+  @SuppressWarnings("PMD.CyclomaticComplexity")
+  private ItemEntity createItem(final String... itemProperties)
+      throws InputException, AttributeParseException {
+  
+    validateItemProperties(itemProperties);
+  
+    final ItemParser parser = new ItemParserImpl();
+    String itemName = "";
+    double itemPrice = 0;
+    double itemQuantity = 0;
+    ItemType itemType = ItemType.RAW;
+    
+    for (int index = 0; index < itemProperties.length; index += 2) {
+      try {
+        switch (itemProperties[index]) {
+          case FlagsConstantsUtils.NAME_FLAG:
+            itemName = parser.parseName(itemProperties[index + 1]);
+            break;
+          case FlagsConstantsUtils.PRICE_FLAG:
+            itemPrice = parser.parsePrice(itemProperties[index + 1]);
+            break;
+          case FlagsConstantsUtils.QUANTITY_FLAG:
+            itemQuantity = parser.parseQuantity(itemProperties[index + 1]);
+            break;
+          case FlagsConstantsUtils.TYPE_FLAG:
+            itemType = parser.parseType(itemProperties[index + 1]);
+            break;
+          default:
+            throw new InputException(
+                String.format(ExceptionsConstantsUtils.INVALID_TYPE, itemProperties[index]));
+        }
+      } catch (IndexOutOfBoundsException exception) {
+        logger.error(
+            String.format(ExceptionsConstantsUtils.INVALID_INPUT_NO_DATA_FOR_FLAG,
+            itemProperties[index]), exception);
+        throw new InputException(
+            String.format(ExceptionsConstantsUtils.INVALID_INPUT_NO_DATA_FOR_FLAG,
+            itemProperties[index]), exception);
+      } catch (AttributeParseException exception) {
+        if (exception.getMessage().equals(
+            String.format(ExceptionsConstantsUtils.INVALID_INPUT_DATA_NOT_FLAG,
+            itemProperties[index + 1]))) {
+          logger.error(String.format(ExceptionsConstantsUtils.INVALID_INPUT_NO_DATA_FOR_FLAG,
+              itemProperties[index]), exception);
+          throw new InputException(
+            String.format(ExceptionsConstantsUtils.INVALID_INPUT_NO_DATA_FOR_FLAG,
+            itemProperties[index]), exception);
```

Comment:
remove this after fixing the error message in ItemParserImplementation.

Reply:
done.